packages/connect-core/src/serialization.ts

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import type {
+  BinaryReadOptions,
+  BinaryWriteOptions,
+  JsonReadOptions,
+  JsonWriteOptions,
+  Message,
+  MessageType,
+  MethodInfo,
+  PartialMessage,
+} from "@bufbuild/protobuf";
+import { ConnectError } from "./connect-error.js";
+import { Code } from "./code.js";
+
+/**
+ * Serialization provides methods to serialize or parse data with a certain
+ * format.
+ */
+export interface Serialization<T> {
```

Comment:
Dumb comment, ignore me if you want, but I feel like `Serializable` makes more sense as the interface name.  Maybe I'm still hung up on Java's Serializable interface though.

Reply:
An object that implements `interface Serializable` is serializable. But this isn't used by objects that are serializable, it's just an object that has two methods `parse()` and `serialize()`, so I don't think `Serializable` fits. 

Not a huge fan of the name though and open to suggestions.

Reply:
Going to merge this because there's some work in the pipeline based on this. Let's discuss offline.